skgstat/Kriging.py

```diff
@@ -386,24 +405,38 @@ def _krige(self, idx):
             t0 = time.time()
 
         p = self.transform_coordinates[idx,:]
-        dists = self.transform_dists[idx,:]
+        if isinstance(self.transform_dists, scipy.sparse.spmatrix):
+            dists = self.transform_dists.getrow(idx)
+        else:
+            dists = self.transform_dists[idx,:]
         
         # find all points within the search distance
-        idx = np.where(dists <= self.range)[0]
+        if isinstance(dists, scipy.sparse.spmatrix):
+            idx = np.array([k[1] for k in dists.todok().keys()])
+        else:
+            idx = np.where(dists <= self.range)[0]
 
         # raise an error if not enough points are found
         if idx.size < self._minp:
             raise LessPointsError
 
         if idx.size > self._maxp:
-            sorted_idx = np.argsort(dists)
-            idx = sorted_idx[np.isin(sorted_idx, idx)][:self._maxp]
-
+            if isinstance(dists, scipy.sparse.spmatrix):
+                selected_dists = dists[0, idx].toarray()[0,:]
+            else:
+                selected_dists = dists[idx]
+            sorted_idx = np.argsort(selected_dists, kind="stable")
```

Comment:
just out of curiosity: why stable sort here?

Reply:
Because without that I couldn't do a regression test that gave the same result with sparse/non-sparse (when point coords coincided exactly for some pair of points, which some did in my test dataset)..